File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.py
from collections import deque

def solution(l, r):
    answer = []
    arr = []
    que = deque(['5'])
    while que:
        num = que.popleft()
        if l <= int(num) <= r: answer.append(int(num))
        
        if int(num+'0') <= r: que.append(num+'0')
        if int(num+'5') <= r: que.append(num+'5')
    
    if not len(answer): answer.append(-1)
    return answer

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>
#include <iostream>
using namespace std;

vector<int> solution(int l, int r) {
    // 1 <= l <= r <= 1e6
    vector<int> answer;
    deque<int> que = {5};
    int num = 0;
    while (num <= r){
        num = que[0];
        que.pop_front();
        if (l <= num && num <= r) answer.push_back(num);
        que.push_back(num*10);
        que.push_back(num*10+5);
    }
    if (answer.size() == 0) answer.push_back(-1);
    
    return answer;
}